### DISCUSSION TITLE
Allow setting deep limit when exporting in app

### DISCUSSION THREAD

**Discussion by sarttiso** [directus/directus#16175] [Feature Requests]
### Describe the Bug

Consider a collection with a field that is an M2M relationship, and that an item or items of this collection contain several hundred linked items via the M2M relationship. When exporting from this collection via the app, only 100 of the linked items will be exported as JSON. This is true regardless of the export format (csv, JSON, XML)), export location (file library or download), and limit specified for export. 

### To Reproduce

1. Generate a collection with several hundred entries.
2. Generate another collection with a field that is an M2M relationship with the first collection.
3. Create an item in the second collection that contains more than 100 linked items in the M2M field.
4. Export the second collection.
5. Inspect the export field containing the JSON from the M2M relationship and find that only 100 items are contained.

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

16.17.0

### What database are you using?

PostGIS 13

### What browser are you using?

Firefox

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
Not quite sure whether to treat this as a bug "always limits at 100" or as a feature request/improvement "allow setting deep limit from app" 🤔 It's technically working as expected/designed, but agree that this is an improvement that should be made

**Comment by azrikahar** [comment]
> Not quite sure whether to treat this as a bug "always limits at 100" or as a feature request/improvement "allow setting deep limit from app" 🤔 It's technically working as expected/designed, but agree that this is an improvement that should be made

Definitely agree that it is a potential improvement! Was tagging `Bug (Potential)` just to be sure it's confirmed. That being said, I do also wonder how would we allow such configuration from a UI perspective. Perhaps it revolves around supporting `deep` in App filters in general first? Or maybe the limit can be "shared" as a default (as in when we set limit `-1` on the parent, the app will auto add the same deep limit for relational fields), before it's made configurable. Just thinking out loud here 🤔

Moving this to discussion while we're exploring potential solution for this improvement 👍

**Comment by sarttiso** [comment]
Wish I could be more helpful with suggesting solutions, but it would be a very useful improvement to be able to export the full list of linked items in the M2M relationship!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
